src/lua/util.rs

```diff
@@ -317,3 +321,50 @@ pub fn to_yaml<'a>(util: Table<'a>, lua: &Lua) -> Result<Table<'a>> {
     util.set("to_yaml", func)?;
     Ok(util)
 }
+
+/// Get a style object for the given path
+///
+/// Type: function( path ) -> style
+///
+/// Example:
+///
+/// ```lua
+/// xplr.util.lscolor("Desktop")
+/// -- { fg = "Red", bg = nil, add_modifiers = {}, sub_modifiers = {} }
+/// ```
+pub fn lscolor<'a>(util: Table<'a>, lua: &Lua) -> Result<Table<'a>> {
+    let lscolors = LsColors::from_env().unwrap_or_default();
+    let func = lua.create_function(move |lua, path: String| {
+        let style = lscolors
+            .style_for_path(path)
+            .map_or(Style::default(), Style::from);
+        lua::serialize(lua, &style).map_err(LuaError::custom)
+    })?;
+    util.set("lscolor", func)?;
+    Ok(util)
+}
+
+/// Format a string using a style object
+///
+/// Type: function( string, style ) -> string
+///
+/// Example:
+///
+/// ```lua
+/// xplr.util.style("Desktop", { fg = "Red", bg = nil, add_modifiers = {}, sub_modifiers = {} })
+/// -- "\u001b[31mDesktop\u001b[0m"
+/// ```
+pub fn style<'a>(util: Table<'a>, lua: &Lua) -> Result<Table<'a>> {
+    let func =
+        lua.create_function(|lua, (string, style): (String, Option<Table>)| {
+            let style: Style = if let Some(s) = style {
+                lua.from_value(Value::Table(s))?
+            } else {
+                Style::default()
+            };
+            let ansi_style: nu_ansi_term::Style = style.into();
+            Ok::<String, LuaError>(ansi_style.paint(string).to_string())
+        })?;
+    util.set("style", func)?;
+    Ok(util)
+}
```

Comment:
Let's also handle NO_COLOR. `use crate::ui::NO_COLOR`. Also, instead of the name `style`, I think `paint` sounds better.

Reply:
Also, let's avod unnecessary computation (i.e. return string) if style is `nil`.

Reply:
One thing I wasn't quite sure about, should we return the input string early in case of NO_COLOR to maximize performance or still try to parse the input into a Style object which could potentially fail to consistently give the same error with the same input?

Reply:
Yeah, that's also what I went with now. Just wanted to make sure